webapp-django/crashstats/api/views.py

```diff
@@ -151,18 +152,22 @@ def no_csrf_i_mean_it(fun):
     @wraps(fun)
     def _no_csrf(request, *args, **kwargs):
         ret = fun(request, *args, **kwargs)
+
         # Remove any csrf bits from Django or django-session-csrf so they don't persist
         if hasattr(request, "_anon_csrf_key"):
             del request._anon_csrf_key
+
+        if "csrf_token" in request.session:
+            del request.session["csrf_token"]
         return ret
 
     return _no_csrf
 
 
 @anonymous_csrf_exempt
 @csrf_exempt
-@no_csrf_i_mean_it
 @clear_empty_session
+@no_csrf_i_mean_it
```

Comment:
We need `no_csrf_i_mean_it` to run and remove the `csrf_token` that django adds to the session. That way the session will get cleared.